app/mapping/ShareAssetMapper.scala

```diff
@@ -56,8 +57,11 @@ class ShareAssetMapper extends Mapper[SharesType] {
         } and
         onStockExchange(SharesOnStockExchangePage) and
         ShareValueInTrustPage.path.read[Long] and
-        Reads(_ => JsSuccess(Some(false)))
-    ) (SharesType.apply _)
+        Reads(_ => JsSuccess(Some(false))) and
+        ShareClassPage.path.read[ShareClass].flatMap {
+          case shareClassValue => Reads(_ => JsSuccess(Some(shareClassValue)))
+        }
```

Comment:
I believe this can just be `ShareClassPage.path.read[ShareClass].map(Some(_))`

Reply:
Good point. Updated